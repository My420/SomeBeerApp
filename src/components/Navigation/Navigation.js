import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import ListGroup from '../ListGroup/ListGroup';
import ListGroupItem from '../ListGroupItem/ListGroupItem';
import styles from './Navigation.module.scss';
import Cart from '../CartLink/CartLink';
import { CART } from '../../utils/constants';

const Navigation = ({ links }) => {
  return (
    <nav className={styles.navigation}>
      <ListGroup className={styles.list}>
        {links.map(element => {
          return (
            <ListGroupItem key={element[0]} className={styles.item}>
              <NavLink
                to={element[0]}
                className={styles.link}
                activeClassName={styles.active}
                exact
              >
                {element[1] === CART ? <Cart /> : element[1]}
              </NavLink>
            </ListGroupItem>
          );
        })}
      </ListGroup>
    </nav>
  );
};

Navigation.propTypes = {
  links: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)).isRequired
};

export default Navigation;
