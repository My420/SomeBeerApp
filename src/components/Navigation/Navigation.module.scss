@import '../../sass/variables.scss';

.list {
  display: flex;
  justify-content: space-around;
  width: $PC-container-width;
  list-style: none;
}

.link {
  font-size: 22px;
  font-weight: 700;
  color: $menu-color;

  text-decoration: none;
  text-transform: uppercase;
}

.link:hover,
.active {
  color: $menu-hover-color;
}

.link:active {
  opacity: 0.5;
}

.link:focus {
  outline: 2px solid $menu-hover-color;
}

.list .item:first-child .active {
  color: $menu-color;
}

.list .item:last-child {
  margin-left: 200px;
}

@media (max-width: $PC-max-width) {
  .list {
    width: $MPC-container-width;
  }

  .list .item:last-child {
    margin-left: 100px;
  }
}

@media (max-width: $MPC-max-width) {
  .navigation {
    position: absolute;
    bottom: -420px;

    width: 100%;

    background-color: $menu-bar-color;
  }

  .list {
    display: block;
    width: auto;
  }

  .list .item:last-child {
    margin-left: 0px;
  }

  .item {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 60px;
  }
}

@media (max-width: $Tablet-max-width) {
  .navigation {
    bottom: -350px;
  }

  .link {
    font-size: 18px;
  }

  .item {
    height: 50px;
  }
}

@media (max-width: $Mobile-max-width) {
}
