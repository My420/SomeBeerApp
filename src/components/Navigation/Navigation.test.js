import React from 'react';
import { shallow } from 'enzyme';
import Navigation from './Navigation';

describe('Test <Navigation /> component', () => {
  const testLinks = [['/', 'Main'], ['/test', 'Test']];

  const wrapper = shallow(<Navigation links={testLinks} />);

  test('should correct render <Navigation /> component ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render  tag <nav>  with className = navigation', () => {
    expect(wrapper.find('nav.navigation')).toHaveLength(1);
  });

  test('should render <ListGroup/> component  with className = list, inside tag <nav>', () => {
    expect(wrapper.find('nav.navigation').find('ListGroup')).toHaveLength(1);
    expect(wrapper.find('ListGroup').prop('className')).toBe('list');
  });

  test('should render 2 <ListGroupItem/> component  with className = item, inside tag <ListGroup/>', () => {
    expect(wrapper.find('ListGroup').find('ListGroupItem')).toHaveLength(2);
    expect(
      wrapper
        .find('ListGroupItem')
        .first()
        .prop('className')
    ).toBe('item');
    expect(
      wrapper
        .find('ListGroupItem')
        .last()
        .prop('className')
    ).toBe('item');
  });

  test('should render 2 <ListGroupItem/> component  with correct key props', () => {
    expect(
      wrapper
        .find('ListGroupItem')
        .first()
        .key()
    ).toBe(testLinks[0][0]);
    expect(
      wrapper
        .find('ListGroupItem')
        .last()
        .key()
    ).toBe(testLinks[1][0]);
  });

  test('should render corect <NavLink/> component, inside tag <ListGroupItem/>', () => {
    expect(
      wrapper
        .find('ListGroupItem')
        .first()
        .find('NavLink')
        .prop('to')
    ).toBe(testLinks[0][0]);

    expect(
      wrapper
        .find('ListGroupItem')
        .first()
        .find('NavLink')
        .prop('children')
    ).toBe(testLinks[0][1]);

    expect(
      wrapper
        .find('ListGroupItem')
        .last()
        .find('NavLink')
        .prop('to')
    ).toBe(testLinks[1][0]);

    expect(
      wrapper
        .find('ListGroupItem')
        .last()
        .find('NavLink')
        .prop('children')
    ).toBe(testLinks[1][1]);
  });

  test('should render <NavLink/> with className = link ', () => {
    expect(
      wrapper
        .find('NavLink')
        .first()
        .prop('className')
    ).toBe('link');

    expect(
      wrapper
        .find('NavLink')
        .last()
        .prop('className')
    ).toBe('link');
  });
});
