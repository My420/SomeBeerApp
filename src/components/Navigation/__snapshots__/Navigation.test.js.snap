// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Navigation /> component should correct render <Navigation /> component  1`] = `
<nav
  className="navigation"
>
  <ListGroup
    className="list"
    tag="ul"
  >
    <ListGroupItem
      className="item"
      key="/"
    >
      <NavLink
        activeClassName="active"
        className="link"
        exact={true}
        to="/"
      >
        Main
      </NavLink>
    </ListGroupItem>
    <ListGroupItem
      className="item"
      key="/test"
    >
      <NavLink
        activeClassName="active"
        className="link"
        exact={true}
        to="/test"
      >
        Test
      </NavLink>
    </ListGroupItem>
  </ListGroup>
</nav>
`;
